File: Backend/C#/Backend/Models/Order.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Backend.Validators;
using Newtonsoft.Json;

namespace Backend.Models {
	public class Order {
		
		public int Id { get; set; }
		
		[Required]
		[Name]
		[MaxLength(63)]
		[MinLength(2)]
		public string FirstName { get; set; }
		
		[Required]
		[Name]
		[MaxLength(63)]
		[MinLength(2)]
		public string LastName { get; set; }
		
		[Required]
		[PostalCode]
		[MaxLength(7)]
		public string PostalCode { get; set; }
		
		[Required]
		[HouseNr]
		[MaxLength(5)]
		public string HouseNr { get; set; }
		
		[Required]
		[Name]
		[MaxLength(63)]
		[MinLength(2)]
		public string StreetName { get; set; }
		
		[Required]
		[Name]
		[MaxLength(63)]
		[MinLength(2)]
		public string Residence { get; set; }
		
		[Required]
		[Phone]
		[MaxLength(12)]
		public string PhoneNr { get; set; }
		
		[Required]
		[MinValue(0)]
		public int TotalPrice { get; set; }
		
		public DateTime CurDate { get; set; }
		
		[Required]
		[UserExists]
		public int UserId { get; set; }
		
		[JsonIgnore]
		[ForeignKey("UserId")]
		public User User { get; set; }
		
		public Order() {}

		public Order(string firstName, string lastName, string postalCode, string houseNr, string streetName,
			string residence, string phoneNr, int totalPrice, int userId, int id = 0) {
			Id = id;
			FirstName = firstName;
			LastName = lastName;
			PostalCode = postalCode;
			HouseNr = houseNr;
			StreetName = streetName;
			Residence = residence;
			PhoneNr = phoneNr;
			TotalPrice = totalPrice;
			UserId = userId;
		}

		public override string ToString() {
			return $"{nameof(Id)}: {Id}, {nameof(FirstName)}: {FirstName}, {nameof(LastName)}: {LastName}" +
			       $", {nameof(PostalCode)}: {PostalCode}, {nameof(HouseNr)}: {HouseNr}" +
			       $", {nameof(StreetName)}: {StreetName}, {nameof(Residence)}: {Residence}" +
			       $", {nameof(PhoneNr)}: {PhoneNr}, {nameof(TotalPrice)}: {TotalPrice}, {nameof(CurDate)}: {CurDate}" +
			       $", {nameof(UserId)}: {UserId}";
		}
	}
}

File: Backend/java/SpringBoot/src/main/java/nl/bookshop/models/Order.java
package nl.bookshop.models;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.*;
import nl.bookshop.validators.*;

import javax.persistence.*;
import javax.validation.constraints.Min;
import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;
import java.sql.Timestamp;

import static nl.bookshop.models.ModelStrings.*;
import static nl.bookshop.validators.ValidationStrings.NEGATIVE_TOTAL_PRICE;

@Data
@AllArgsConstructor
@RequiredArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = ORDERS)
public class Order {

    @NonNull
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @NonNull
    @Name
    @NotEmpty
    private String firstName;

    @NonNull
    @Name
    @NotEmpty
    private String lastName;

    @NonNull
    @PostalCode
    @NotEmpty
    private String postalCode;

    @NonNull
    @HouseNr
    @NotEmpty
    private String houseNr;

    @NonNull
    @Name
    @NotEmpty
    private String streetName;

    @NonNull
    @Name
    @NotEmpty
    private String residence;

    @NonNull
    @PhoneNr
    @NotEmpty
    private String phoneNr;

    @NonNull
    @NotNull
    @Min(value = 0, message = NEGATIVE_TOTAL_PRICE)
    private Integer totalPrice;


    @NonNull
    @NotNull
    @UserExists
    private Integer userId;

    @JsonIgnore
    private Timestamp curDate = new Timestamp(System.currentTimeMillis());
}
